source/material_model/depth_dependent.cc

```diff
@@ -0,0 +1,444 @@
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    DepthDependent<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection("Depth dependent model");
+        {
+          base_model.reset(create_material_model<dim>(prm.get("Base model")));
+          if ( prm.get("Depth dependence method") == "Function" )
+            {
+              viscosity_method = Function;
+            }
+          else if ( prm.get("Depth dependence method") == "File" )
+            {
+              viscosity_method = File;
+            }
+          else if ( prm.get("Depth dependence method") == "List" )
+            {
+              viscosity_method = List;
+            }
+          else if (  prm.get("Depth dependence method") == "None" )
+            {
+              viscosity_method = None;
+            }
+          else
+            {
+              Assert(false, ExcMessage("Unknown method for depth dependence."));
+            }
+
+          depth_values     = Utilities::string_to_double(Utilities::split_string_list(prm.get("Depth list")));
+          viscosity_values = Utilities::string_to_double(Utilities::split_string_list(prm.get("Viscosity list")));
+          /*
+           * check sanity of viscosity list values and depth list values input
+           */
+          if ( viscosity_method == List)
+            {
+              /* check that length of depth values and viscosity values are compatible */
+              Assert( depth_values.size() == viscosity_values.size() ,
+                      ExcMessage("Depth list must be same size as Viscosity list"));
+              /* check that list is in ascending order */
+              for ( int i=1; i<depth_values.size(); i++)
+                Assert(depth_values[i] > depth_values[i-1],
+                       ExcMessage("Viscosity depth values must be strictly ascending"));
+              /* check that last layer includes base of model */
+              const double maximal_depth = this->get_geometry_model().maximal_depth();
+              Assert( *(depth_values.end()-1) >= maximal_depth,
+                      ExcMessage("Last value in Depth list must be greater than or equal to maximal depth of domain"));
+            }
+
+          datadirectory                = prm.get ("Data directory");
+          {
+            const std::string      subst_text = "$ASPECT_SOURCE_DIR";
+            std::string::size_type position;
+            while (position = datadirectory.find (subst_text),  position!=std::string::npos)
+              datadirectory.replace (datadirectory.begin()+position,
+                                     datadirectory.begin()+position+subst_text.size(),
+                                     ASPECT_SOURCE_DIR);
+          }
+          radial_viscosity_file_name   = prm.get ("Viscosity depth file");
```

Comment:
I think you should be able to parse the input file here already if that's what the user selected. In that case, you also don't have to store the directory and file names in the class -- just make them local variables here.